src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -383,24 +383,28 @@ def _await_fluent_launch(
 
 
 def _get_server_info(
-    server_info_file_name: str,
+    server_info_file_name: Optional[str] = None,
     ip: Optional[str] = None,
     port: Optional[int] = None,
     password: Optional[str] = None,
 ):
     """Get server connection information of an already running session."""
-    if ip and port:
+    if not (ip and port) and not server_info_file_name:
+        raise RuntimeError(
+            "Please provide either ip and port data or server-info file."
+        )
+    if ip and port and server_info_file_name:
         logger.warning(
             "Could not parse server-info file because ip and port were provided explicitly."
         )
-    elif server_info_file_name:
-        ip, port, password = _parse_server_info_file(server_info_file_name)
-    elif os.getenv("PYFLUENT_FLUENT_IP") and os.getenv("PYFLUENT_FLUENT_PORT"):
-        ip = port = None
     else:
-        raise RuntimeError(
-            "Please provide either ip and port data or server-info file."
-        )
+        if server_info_file_name:
+            ip, port, password = _parse_server_info_file(server_info_file_name)
+        ip = ip or os.getenv("PYFLUENT_FLUENT_IP", "127.0.0.1")
+        port = port or os.getenv("PYFLUENT_FLUENT_PORT")
+
+    if not port:
+        raise ValueError("Port needs to be specified.")
```

Comment:
@prmukherj CC: @mkundu1 @raph-luc 

Another key improvement would be to ensure that we add the full context of the error. But please again note the aforementioned PR from @hpohekar.

Reply:
Done. https://github.com/ansys/pyfluent/pull/2176/files#diff-00c898f4e09231f8a736f0d6a5d8903afe4b6e4ae8a8a12b30b9234f8d70bddb.

Reply:
@prmukherj can remove this change now.